docs/book/src/multiversion-tutorial/tutorial.md

```diff
@@ -0,0 +1,294 @@
+# Implement multi-version API
+
+This tutorial will teach you how to build an API with multiple versions.
+
+## Concepts
+Before we dive further, let's take a quick look at some of the basic concepts 
+related to multiple versions in Kubernetes.
+
+### Hub/Spoke/Storage Versions
+All versions of a Kubernetes object share the same underlying storage. So if you
+have versions v1, v2 and v3 of a kind, kubernetes will use one of the versions to
+persist the object of that Kind in Etcd. You can specify the version to be used
+for storage in the Custom Resource definition for that API.
+
+One can think of storage version as the hub and other versions as spokes to visualize the
+relationship between storage and other versions (as shown below in the diagram).
+The important thing to note is that conversion between storage and other version
+should be lossless (round trippable). As shown in the diagram below, v3 is the 
+storage/hub version and v1 and v2 spoke version. This tutorial uses 
+the terms `storage version` and `hub` interchangeably.
```

Comment:
We should explain briefly how does converting spoke to spoke works.

Reply:
Done.